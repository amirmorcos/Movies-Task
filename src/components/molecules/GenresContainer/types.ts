import { StyleProp, ViewStyle } from "react-native";

export type GenresContainerProps = {
  genres: string[];
  overrideContainerStyle?: StyleProp<ViewStyle>;
};
