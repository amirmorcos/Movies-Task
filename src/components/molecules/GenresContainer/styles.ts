import { StyleSheet } from "react-native";
import { moderateScale, s, vs } from "react-native-size-matters";

export default StyleSheet.create({
  container: {
    backgroundColor: "white",
    paddingHorizontal: s(5),
    paddingVertical: vs(2),
    width: s(100),
    justifyContent: "center",
    alignItems: "center",
  },
  genre: {
    fontSize: moderateScale(10),
  },
});
