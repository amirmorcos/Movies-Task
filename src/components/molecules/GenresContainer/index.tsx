import React from "react";
import { Text, View } from "react-native";
import styles from "./styles";
import { GenresContainerProps } from "./types";

const GenresContainer = ({
  genres,
  overrideContainerStyle,
}: GenresContainerProps) => {
  return (
    <View style={[styles.container, overrideContainerStyle]}>
      <Text numberOfLines={1} ellipsizeMode="tail" style={styles.genre}>
        {genres}
      </Text>
    </View>
  );
};

export default GenresContainer;
