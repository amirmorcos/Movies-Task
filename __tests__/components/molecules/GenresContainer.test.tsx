import GenresContainer from "molecules/GenresContainer";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("react-native-vector-icons/FontAwesome6", () => "Icon");
jest.mock("react-native-size-matters", () => ({
  s: jest.fn(),
  vs: jest.fn(),
  moderateScale: jest.fn(),
}));

test("renders correctly", () => {
  const tree = renderer
    .create(<GenresContainer genres="Action, Romance" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
