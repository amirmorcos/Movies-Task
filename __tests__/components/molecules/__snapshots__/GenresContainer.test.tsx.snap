// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "justifyContent": "center",
        "paddingHorizontal": undefined,
        "paddingVertical": undefined,
        "width": undefined,
      },
      undefined,
    ]
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "fontSize": undefined,
      }
    }
  >
    Action, Romance
  </Text>
</View>
`;
